doc/release/upcoming_changes/13458.changes.rst

```diff
@@ -0,0 +1,4 @@
+Update the documentation for `__array__` interface behaviour 
+--------------------------------------------------
+Earlier the `__array__` documentation said:
+If a class (ndarray subclass or not) having the `__array__` method is used as the output object of an ufunc, results will be written to the object returned by `__array__` However the behaviour of `__array__` did not support it, the documentation was modified to avoid confusion.
```

Comment:
Thanks for trying to be thorough - but there's no need for release notes for documentation changes.

Reply:
Thank you @eric-wieser. I have made the suggested changes.